container.go

```diff
@@ -422,6 +327,11 @@ func newContainer(pod *Pod, contConfig ContainerConfig) (*Container, error) {
 		return &Container{}, fmt.Errorf("Invalid container configuration")
 	}
 
+	devices, err := newDevices(contConfig.DeviceInfos)
```

Comment:
This is gonna be called all the time, even when we don't need it (when we can actually retrieve devices from the runtime storage). Instead I think we should call this only if we cannot retrieve anything from storage, OR the other solution is to make the difference between the first time this container has been created and all other times when it is only recreated.

Reply:
@sboeuf This is what I was doing earlier, but changed it later.
@sameo Giving it a bit more thought, I like the earlier approach of creating the devices only when fetch from storage was not successful.

Reply:
@amshinde It's more efficient, I agree. We can keep it there, but please add some further comments describing why and how we can reach each side of the if switch.

Reply:
Fixed